pkg/cmd/util/variable/variable.go

```diff
@@ -81,7 +81,7 @@ var overrideVersion = version.Get()
 
 // lastSemanticVersion attempts to return a semantic version from the GitVersion - which
 // is either <semver>+<commit> or <semver> on release boundaries.
-func lastSemanticVersion(version string) string {
-	parts := strings.Split(version, "+")
-	return parts[0]
+func lastSemanticVersionWithoutModifiers(version string) string {
+	parts := strings.Split(version, "-")
+	return strings.Split(parts[0], "+")[0]
```

Comment:
If we want to enforce semver, could we be explicit about it by validating the version as proper semver and returning `{major}.{minor}.{patch}` from the parsed version?

Reply:
I believe we decided we did not want to accept an external dependency on a semver library in this code.